### ISSUE TITLE
Support for old-style typed namedtuples

### ISSUE THREAD

**Issue by PeterJCLaw** [davidhalter/jedi#1532]
jedi has some support for untyped `namedtuple`s` and for typed `NamedTuple`s which are constructed as classes, however doesn't seem to do much for typed `NamedTuple`s which are spelled using the legacy construction:

``` python
TNT = NamedTuple('TNT', (
    ('value', int),
))
```

I appreciate that this style is becoming less popular now that Python 3.5 is getting quite old, though there's probably still a lot of code which uses it.

Since there does seem to be support for untyped named tuples:
``` python
NT = namedtuple('NT', (
    'value',
))

nt = NT(value=42)
print(nt.value)
#        ^ you do get completions here
```

I'm hoping that this would be relatively easy to add.

My original use case was hoping for completions from a function annotated as returning a typed named tuple, though I think the issue comes down to the named tuple being declared using the older syntax (the codebase I was in is Python 3.5).

**Comment by davidhalter** [comment]
> I'm hoping that this would be relatively easy to add.

It's definitely possible to add it, but it's probably still quite a bit of work. I would not recommend doing it, just because I'm dropping Python 2/3.5 very, very soon. However since people might still use this even in newer Python versions - I will not stop you from implementing it :).

**Comment by PeterJCLaw** [comment]
Ah, I did wonder if 2/3.4/3.5 were going to be dropped soon. (Unfortunately I have some codebases stuck on 3.5 for the moment, so I'm hoping not too soon).

I think I have a halfway working version of this locally (a light modification of the `collections_namedtuple` function, so I might spend a little more time at some point getting that to work.

**Comment by davidhalter** [comment]
> so I'm hoping not too soon

The old Jedi versions will keep working :)

**Comment by davidhalter** [comment]
I'm closing, since I won't implement it and don't think it's necessary. But feel free to ask questions here or open a PR.

**Comment by PeterJCLaw** [comment]
So where I got to was that I'm modifying the existing named tuple template to add type information, which works well enough when I hand edit, but less so programatically.

The issue seems to be that any names for types that aren't builtins aren't available when we parse the new type.

I can see we've got an inference state and a context, so I feel like I want to either:
- extract the names from those and make them available to our new parsing state/context, or
- somehow inject the new code into the existing context

The latter feels like it could get quite messy, so I'm assuming the former is the way to go. Is there a way to get the names from one state/context and insert them into another as a kind of bootstrap?
